week-4/Homework/mandatory/3-water-bottle.js

```diff
@@ -1,41 +1,187 @@
 /*
-Create an object that acts like a water bottle. 
-It will need a volume key to store how full or empty the bottle is. 
-It will be 100 when full and 0 when empty. 
-Give your water bottle methods for filling it up, 
-drinking some of it, and emptying it.
+Create an object that acts a water bottle.
 
-We made a start on this for you here: 
+It will need a volume property to store how full or empty the bottle is. 
+Volume will be 100 when bottle is full and 0 when empty. 
+
+Give your water bottle methods for 
+  - filling it up
+  - pouring 10 unit of water into it
+    Note: You cannot over exceed the bottle capacity.
+  - drinking 10 unit from it
+    Note: You cannot drink more than its actual content.
+  - and telling if the bottle is full
+  - and telling if the bottle is empty
+
+We made a start on this for you here by giving the sceleton of our object.
+You have to implement the missing features according to the specification.
 */
 
+// Here is your starting point:
 let bottle = {
   volume: 0,
-  fill: function() {
-    // calling this function should make you bottles volume = 100;
+  fillUp: function () {
+    // calling this function should pour your bottle full (volume = 100);
+    return (this.volume = 100);
+  },
+
+  pour: function () {
+    // calling this function should increase your bottle volume by 10 unit;
+    this.volume <= 90 ? (this.volume += 10) : this.volume;
+  },
+  drink: function () {
+    // calling this function should decrease your bottle volume by 10 unit;
+    this.volume >= 10 ? (this.volume -= 10) : this.volume;
+  },
+  isFull: function () {
+    // this function should return true if your bottle is empty;
+    return this.volume == 100 ? true : false;
   },
-  drink: function() {
-    // calling this function should decrease your bottles volume by 10;
+  isEmpty: function () {
+    // this function should return true if your bottle is full;
+    return this.volume == 0 ? true : false;

```

Comment:
```suggestion
    return this.volume === 0;
```
Same issue here in that you only to `return` the expression as it will evaluate to a boolean.